### PR TITLE
DECIMAL_STR_WIDTH() + DECIMAL_STR_MAX() tweaks

### PR DESCRIPTION
**Pull request by poettering**
Inspired by e3dd9ea8ea4510221f73071ad30ee657ca77565d

### CODE REVIEW

File: src/basic/macro.h [hunk 1]
```diff
@@ -309,22 +309,31 @@ static inline int __coverity_check_and_return__(int condition) {
...
         ({                                                        \
                 typeof(x) _x_ = (x);                              \
-                size_t ans = IS_SIGNED_INTEGER_TYPE(_x_) ? 2 : 1; \
-                while ((_x_ /= 10) != 0)                          \
-                        ans++;                                    \
+                size_t ans;                                       \
+                if (_x_ == 0)                                     \
+                        ans = 1;                                  \
+                else {                                            \
+                        ans = _x_ <= 0 ? 2 : 1;                   \
```
Comment (mrc0mmand) [inline src/basic/macro.h:333]: D'oh, neat!

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/basic/macro.h [hunk 2]
```diff
@@ -309,15 +309,14 @@ static inline int __coverity_check_and_return__(int condition) {
...
+/* Returns the number of chars needed to format variables of the specified type as a decimal string. Adds in
+ * extra space for a negative '-' prefix for signed types. Includes space for the trailing NUL. */
 #define DECIMAL_STR_MAX(type)                                           \
-        (2U+(sizeof(type) <= 1 ? 3U :                                   \
+        ((size_t) IS_SIGNED_INTEGER_TYPE(type) + 1U +                   \
+            (sizeof(type) <= 1 ? 3U :                                   \
              sizeof(type) <= 2 ? 5U :                                   \
              sizeof(type) <= 4 ? 10U :                                  \
-             sizeof(type) <= 8 ? 20U : sizeof(int[-2*(sizeof(type) > 8)])))
+             sizeof(type) <= 8 ? (IS_SIGNED_INTEGER_TYPE(type) ? 19U : 20U) : sizeof(int[-2*(sizeof(type) > 8)])))
```
Comment (medhefgo) [inline src/basic/macro.h:319]: This is confusing me. Why only care about signed for sizeof == 8 while happily over-estimating for the smaller ones? Or am I missing something?

File: src/test/test-macro.c [hunk 3]
```diff
@@ -410,4 +410,27 @@ TEST(decimal_str_width) {
         assert_se(DECIMAL_STR_WIDTH(INT64_MIN) == STRLEN("-9223372036854775808"));
 }
 
+TEST(decimal_str_max) {
```
Comment (medhefgo) [inline src/test/test-macro.c]: Doesn't really matter, but so far, uppercase macros also had their test name uppercase. (Goes for the decimal_str_width test too)

File: src/basic/macro.h (hunk 2 above)
Comment (poettering) [reply src/basic/macro.h:319]: the commit msg is supposed to explain that...

1. INT8_MIN and UINT8_MAX (as the "longest" decimal numbers fitting in 8 bit) both have three digits (only talking about digits here, ignore the "-" for now, it is handled alreay in the first component of the expression)
2. similar, INT16_MIN and UINT16_MAX have 5 digits
3. INT32_MIN and UINT32_MAX have 10 digits.
4. but INT64_MIN and UINT64_MAX have a different number of digits! INT64_MIn as 19 and UINt64_MAX has 20! hence the special casing.

Comment (medhefgo) [reply src/basic/macro.h:319]: I see. now it makes sense. But that really doesn't come across in the commit message (to me).

Comment (poettering) [reply src/basic/macro.h:319]: i extended the commit msg a bit

**Pull request merged**